### PR TITLE
Irqv5

### PR DESCRIPTION
**Pull request by lyan3**
This two commits remove the irq sharing support,  main changes including:
 - removed the dev_list field in irq_desc, and clean up codes for the list operation; 
 - replace IRQ_ASSIGNED_SHARED and IRQ_ASSIGNED_NOSHARE with IRQ_ASSIGNED;
 - remove "shared" argument in irq APIs;
 - remove dev_handler_node struct and change the return type of pri_/normal_register_handler() from dev_handler_node* to int32_t, which is irq num (>= 0) on success, and errno (> 0) on failure;
 - changes unregister_irq_handler() to take argument unint32_t instead of dev_handler_node*;

In addition,  irq request flow for pt devices is revised, in order to remove dependency on irq sharing:  register irq on adding remapping entery and unregister irq on removal an entry, and do not register/unregister at remapping an entry.

### CODE REVIEW

File: hypervisor/arch/x86/irq.c [hunk 1]
```diff
@@ -151,43 +151,18 @@ static void disable_pic_irq(void)
 	pio_write8(0xffU, 0x21U);
 }
 
-static bool
-irq_desc_append_dev(struct irq_desc *desc, void *node, bool share)
+static void 
+irq_desc_append_dev(struct irq_desc *desc, void *node)
```
Comment (dongyaozu) [inline hypervisor/arch/x86/irq.c]: append irq or dev? We don't have device concept

Comment (dongyaozu) [reply hypervisor/arch/x86/irq.c]: it is more like to register a handler?

Comment (dongyaozu) [reply hypervisor/arch/x86/irq.c]: replace node with action? and should not be void *

Comment (lyan3) [reply hypervisor/arch/x86/irq.c]: This function would be removed completely in the second patch, so I hadn't paid much attention to the name. 
If you insist, I will change the name to set_irq_action(struct irq_desc *desc, void *node)

**Comment by lyan3** [comment]
Hi @dongyaozu , 
Jinxia needs your approval to merge this PR since it has been updated.
Thanks,
Like

**Comment by lijinxia** [comment]
Verified+1

**Pull request merged**